IPv6_test.go

```diff
@@ -153,6 +154,13 @@ func Test_IPv6_String(t *testing.T) {
 		{"1:0:0:0:1:0:0:1", "1::1:0:0:1"},
 		{"1:0:0:0:0:1:0:1", "1::1:0:1"},
 		{"1:0:0:0:0:0:1:1", "1::1:1"},
+		
+		{"1:1:1:1:1:1:1:1", "1:1:1:1:1:1:1:1"},
+		{"1:1:0:1:1:1:1:1", "1:1:0:1:1:1:1:1"}, // see RFC 5952 section 4.2.2
+		{"1:1:0:0:1:1:1:1", "1:1::1:1:1:1"},
+		{"1:1:0:0:0:1:1:1", "1:1::1:1:1"},
+		{"1:1:0:0:0:0:1:1", "1:1::1:1"},
+		{"1:1:0:0:0:0:0:1", "1:1::1"},
```

Comment:
i think that most of these are in some ways a duplicate of the test on line 150 (which tests for  compressing a single span of 6 zero fields in the middle of the addr).  You may adjust these slightly so that they expand on what line 150 is testing a bit and place them below line 150. The test against compressing a single zero field is valid and wasn't being handled properly before. I am submitting a patch for that.

Reply:
If you'll adjust your tests slightly and ensure that they run/pass then I'll accept them.